docs/development/pigs/pig-016.rst

```diff
@@ -0,0 +1,215 @@
+gammapy.cube down to gammapy.background would likely lead to circular
+dependencies, because they rely on the map dataset class in gammapy.cube.
+
+The reflected region background code is not a great fit in gammapy/spectrum or
+anywhere in Gammapy except for gammapy.background, but moving it there does make
+sense if we say that all 1D spectrum code goes in gammapy.spectrum and all
+map-related analysis code goes in gammapy.cube.
+
+Dissolve gammapy.image
+----------------------
+
+The gammapy.image package was introduced early and contained a SkyImage class
+for 2D sky images, and 2D image models, and some image analysis helper functions
+such as code to make a radial profile from an image. Much later gammapy.maps was
+introduced and we managed to support 2D and 3D maps with one container instead
+of separate gammapy.image.SkyImage and gammapy.cube.SkyCube.
+
+Now, if we move gammapy.image.models to gammap.modeling as proposed above, and
+that SkyImage has been removed in 2018, there is very little functionality left
+in gammapy.image, and it's not clear what it's scope is.
+
+So we propose to clean this up, roughly like this:
+
+- asmooth.py and utils.py -> move to gammapy.cube
+- measure.py and profile.py -> replace with docs example or move to maps or cube
+- plotting.py -> replace with docs example or move to maps
+
+Clean up ``gammapy.utils``
+--------------------------
+
+Above we propose to move gammapy.utils.fitting, the biggest piece of
+gammapy.utils to gammapy.modeling. We still need and want to keep gammapy.utils,
+as a lowest-level layer in Gammapy where we write adapters and helpers for
+functionality e.g. from Numpy and Astropy. Mostly this is for usage within
+Gammapy, users very rarely need to import from gammapy.utils.
+
+Checking via ``ack gammapy.utils tutorials | egrep -v fitting`` we see that
+currently only energy_logspace, sqrt_space, sample_powerlaw and
+SphericalCircleSkyRegion are used from end-user documentation, i.e. very little.
+
+We propose to clean up gammapy.utils, e.g. gammapy/utils/coordinates.py still
+contains coordinate helper functions that were added before astropy.coordinates
+existed. We do not detail the cleanup here, this will be discussed on a
+case-by-case basis. After that we will review what is left and decide whether we
+keep or completely remove gammapy.utils from the Gammapy docs, i.e. if it's
+public API or not.
+
+Outlook
+=======
+
+During discussion of this PIG, several other ideas surfaced and were considered.
+We don't propose those changes here and for v1.0, in some cases because
+discussion didn't conclude on whether it's a good idea or not, and in some cases
+because introducing the change would require a detailed dedicated PIG with a
+feasability exploration as in `GH 2290`_ and then significant extra
+developments. We do expect that some of these changes will be made in the
+future (and some won't).
+
+Generalising maps to allow maps without a WCS and spatial axes, and / or using
```

Comment:
I think having one container instead of different ones for spectra, cubes, time axes instead of several containers and datasets could change the package structure. And it was part of the discussion in this PIG (see earlier comment by Karl), and also with your idea to introduce `gammapy.datasets` - that need goes away if there's only 1-2 and they are Map-based, then they would just go in gammapy.maps. So I'll keep this sentence, but clarify: "It's not clear if and what changes this would imply for the Gammapy package structure."